Content.Client/Chat/CollectiveMindChatUpdateSystem.cs

```diff
@@ -0,0 +1,32 @@
+using Content.Client.Chat.Managers;
+using Content.Shared.CollectiveMind;
+using Robust.Client.Player;
+
+namespace Content.Client.Chat
+{
+    public sealed class CollectiveMindChatUpdateSystem : EntitySystem
+    {
+        [Dependency] private readonly IChatManager _chatManager = default!;
+        [Dependency] private readonly IPlayerManager _playerManager = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<CollectiveMindComponent, ComponentInit>(OnInit);
+            SubscribeLocalEvent<CollectiveMindComponent, ComponentRemove>(OnRemove);
+        }
+
+        public CollectiveMindComponent? Player => CompOrNull<CollectiveMindComponent>(_playerManager.LocalPlayer?.ControlledEntity);
+        public bool IsCollectiveMind => Player != null;
```

Comment:
dont store fields in systems like this, specially if you are gonna reference them in another place, make a method.